apps/mobile/src/components/Community/Tabs/CommunityViewPostsTab.tsx

```diff
@@ -0,0 +1,154 @@
+import { Typography } from '~/components/Typography';
+import { CommunityViewPostsTabFragment$key } from '~/generated/CommunityViewPostsTabFragment.graphql';
+import { CommunityViewPostsTabQueryFragment$key } from '~/generated/CommunityViewPostsTabQueryFragment.graphql';
+import { MainTabStackNavigatorProp } from '~/navigation/types';
+
+type Props = {
+  communityRef: CommunityViewPostsTabFragment$key;
+  queryRef: CommunityViewPostsTabQueryFragment$key;
+};
+
+export function CommunityViewPostsTab({ communityRef, queryRef }: Props) {
+  const { data: community } = usePaginationFragment(
+    graphql`
+      fragment CommunityViewPostsTabFragment on Community
+      @refetchable(queryName: "CommunityViewPostsTabFragmentPaginationQuery") {
+        name
+        contractAddress {
+          address
+        }
+        posts(first: $postLast, after: $postBefore)
+          @connection(key: "CommunityViewPostsTabFragment_posts") {
+          edges {
+            node {
+              ...createVirtualizedFeedEventItemsPostFragment
+            }
+          }
+          pageInfo {
+            total
+          }
+        }
+      }
+    `,
+    communityRef
+  );
+
+  const query = useFragment(
+    graphql`
+      fragment CommunityViewPostsTabQueryFragment on Query {
+        ...createVirtualizedFeedEventItemsQueryFragment
+      }
+    `,
+    queryRef
+  );
+
+  const contentContainerStyle = useListContentStyle();
+
+  const totalPosts = community?.posts?.pageInfo?.total ?? 0;
+
+  const { markEventAsFailure, failedEvents } = useFailedEventTracker();
+
+  const posts = [];
+  for (const edge of community?.posts?.edges ?? []) {
+    if (edge?.node) {
+      posts.push(edge.node);
+    }
+  }
+
+  posts.reverse();
+
+  const ref = useRef<FlashList<FeedListItemType> | null>(null);
+  const { items } = createVirtualizedFeedEventItems({
+    itemRefs: posts,
+    listRef: ref,
+    failedEvents,
+    queryRef: query,
+  });
+
+  const renderItem = useCallback<ListRenderItem<FeedListItemType>>(
+    ({ item }) => {
+      // Set a default for feed navigation pill
+      let itemId: string | null = null;
+
+      if (item.post) {
+        itemId = item.post.dbid;
+      } else if (item.event) {
+        itemId = item.event.dbid;
+      } else {
+        itemId = item.eventId;
+      }
+
+      const markFailure = () => (itemId ? markEventAsFailure(itemId) : () => {});
+
+      return <FeedVirtualizedRow item={item} onFailure={markFailure} />;
+    },
+    [markEventAsFailure]
+  );
+
+  const navigation = useNavigation<MainTabStackNavigatorProp>();
+
+  const handleCreatePost = useCallback(() => {
+    if (!community?.contractAddress?.address) return;
+    navigation.navigate('NftSelectorContractScreen', {
+      contractAddress: community?.contractAddress?.address,
+      page: 'Community',
+    });
+  }, [navigation, community?.contractAddress?.address]);
+
+  if (totalPosts === 0) {
```

Comment:
think we need to add the feature flag here, I see this when the flag is off:

for simplicity, on mobile we can keep the tab component and just hide the "Posts" tab if feature is OFF

![Screenshot 2023-08-02 at 10 45 01](https://github.com/gallery-so/gallery/assets/80802871/b1456173-d95f-43f1-950e-cb0f42954edd)

Reply:
Nice catch!

Added the feature flag to hide tab

<img width="462" alt="CleanShot 2023-08-02 at 23 55 29@2x" src="https://github.com/gallery-so/gallery/assets/4480258/47f8d5c2-4450-4786-891d-fa9921cf5eb0">